crates/rust/src/lib.rs

```diff
@@ -230,12 +230,7 @@ impl WorldGenerator for RustWasm {
                         #[used]
                         #[doc(hidden)]
                         #[cfg(target_arch = \"wasm32\")]
-                        static __FORCE_SECTION_REF: fn() = __force_section_ref;
-                        #[doc(hidden)]
-                        #[cfg(target_arch = \"wasm32\")]
-                        fn __force_section_ref() {{
```

Comment:
I've tested the removal of this symbol and it seems like just referencing `__force_section_ref` directly from the `#[used]` static is enough to get the component type custom section linked in.